common/src/str.rs

```diff
@@ -82,6 +82,29 @@ pub fn to_ascii(value: &str) -> String {
     ascii
 }
 
+#[doc(hidden)]
+pub const fn bytes_is_ascii(x: &str) -> bool {
+    let x = x.as_bytes();
+    let mut i = 0;
+    while i < x.len() {
+        if !x[i].is_ascii() {
+            return false;
+        }
+        i += 1;
+    }
+    true
+}
+
+#[macro_export]
+macro_rules! ascii {
+    ($x:literal) => {{
+        const _: () = {
+            ["not ascii"][!::rustpython_common::str::bytes_is_ascii($x) as usize];
+        };
+        unsafe { ::ascii::AsciiStr::from_ascii_unchecked($x.as_bytes()) }
```

Comment:
Oh wait sorry this should probably also be a `$crate::` reference and the `ascii` crate should be `pub use`'d somewhere

Reply:
I created `common::vendored::ascii`